src/functions.php

```diff
@@ -174,9 +174,9 @@
  * $promise->cancel();
  * await($promise);
  * ```
- *
- * @param callable(mixed ...$args):mixed $function
- * @return callable(): PromiseInterface<mixed>
+ * @template T
+ * @param callable(): T $function
+ * @return callable(): PromiseInterface<T>
```

Comment:
Not sure what to do about the `$args`, but it's my understanding the following would be more precise?

```suggestion
 *
 * @template T
 * @param callable(mixed ...$args): T $function
 * @return callable(): PromiseInterface<T>
```